lib/vyasa_web/components/contexts/read.ex

```diff
@@ -888,108 +995,143 @@ defmodule VyasaWeb.Context.Read do
-     the read mode.
-  2. marks:
-     the actual marks state, this strictly follows the invariant that the order
-     of marks kept in the socket here will always be in descending order of their
-     order attribute.
-  3. marks_ui:
-     ui state for the marks, this is currently just for the read mode.
-
-  NOTE: during the init of marks, we will be mutating the draft reflector. This
-  keeps the state of the draft marks on the db-side always sanitised.
+  NOTE that the responsiblity of updating the aux ui struct shall not be owned by this function.
   """
-  def init_marks(
+  def commit_marks_in_reflector(
         %Socket{
           assigns: %{
-            content_action: :show_verses,
-            draft_reflector: draft_reflector
+            draft_reflector: %Vyasa.Sangh.Sheaf{} = curr_sheaf
           }
-        } = socket
+        } = socket,
+        [%Mark{} | _] = new_marks
       ) do
-    IO.puts("INIT_MARKS")
-
-    case draft_reflector do
-      # handles head sheaf with existing marks:
-      %Sheaf{marks: [_ | _] = marks} ->
-        IO.puts("CHECKPOINT A")
-
-        marks_with_draft =
-          [marks |> Mark.get_draft_mark() | marks]
-          |> Mark.sanitise_marks()
-
-        socket
-        |> assign(marks: marks_with_draft)
-        |> assign(marks_ui: marks_with_draft |> MarksUiState.get_initial_ui_state())
-        |> mutate_draft_reflector()
+    sanitised_marks = new_marks |> Mark.sanitise_marks()
 
-      # handles head sheaf without existing marks:
-      %Sheaf{} ->
-        IO.puts("CHECKPOINT B")
-        marks = [Mark.get_draft_mark()]
-
-        socket
-        |> assign(marks: marks)
-        |> assign(marks_ui: marks |> MarksUiState.get_initial_ui_state())
-        |> mutate_draft_reflector()
-    end
-  end
-
-  def init_marks(%Socket{} = socket) do
-    IO.puts("INIT_MARKS POKEMON")
-    marks = [Mark.get_draft_mark()]
+    {:ok, written_sheaf} = Vyasa.Sangh.update_sheaf(curr_sheaf, %{marks: sanitised_marks})
 
     socket
-    |> assign(marks: marks)
-    |> assign(marks_ui: marks |> MarksUiState.get_initial_ui_state())
+    |> assign(draft_reflector: written_sheaf)
+    |> maybe_prepend_draft_mark_in_reflector()
   end
 
+  # inserts an existing verse back into its stream to
+  # trigger dom updates:
   defp cascade_stream_change(
          %Socket{
            assigns: %{
              kv_verses: verses,
              streams: %{verses: _current_verses} = _streams,
-             marks: [%Mark{verse_id: v_id, binding: binding} | _] = _marks
+             draft_reflector:
+               %Sheaf{
+                 marks: [%Mark{} | _] = _marks
+               } = _draft_reflector
            }
          } = socket
        ) do
+    %Verse{
+      id: v_id,
+      binding: binding
+    } =
+      Map.values(verses)
+      |> Enum.find(fn v -> not is_nil(v.binding) end)
+
     socket
     |> stream_insert(
       :verses,
       %{verses[v_id] | binding: binding}
     )
   end
 
```

Comment:
I've changed how this cascade works, to be frank I forgot why the older version would not work with my changes in this PR 🤔 

this new version should be non-breaking anyway